terraform-plans/templates/github/charm_check.yaml.tftpl

```diff
@@ -20,17 +20,60 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  lint-unit:
-    uses: canonical/bootstack-actions/.github/workflows/lint-unit.yaml@v2
+  lint:
+    name: Lint
     strategy:
       fail-fast: false
       matrix:
         python-version: ["3.8", "3.10"]
-    with:
-      python-version: $${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: true
+
+    - name: Set up Python $${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: $${{ matrix.python-version }}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install 'tox<5'
+
+    - name: Run linters
+      run: tox -e lint
+
+  unit:
+    name: Unit
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.10"]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: true
+
+    - name: Set up Python $${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: $${{ matrix.python-version }}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install 'tox<5'
```

Comment:
I think it would be good to add note why we are pining tox <5. WDYT?

Reply:
Sure, good idea, I can do that :+1: 

For the record, the reason is to avoid a repeat of what happened to our CI on every project when tox 4 was originally released.  Safer to pin to a major version, then we can upgrade in a controlled manner when a new major version is released. :sweat_smile:

Reply:
Added :)